### ISSUE TITLE
AWS Lambda - provide context object

### ISSUE THREAD

**Issue by vladaman** [oven-sh/bun#6003]
### What is the problem this feature would solve?

Similar to AWS Lambda for NodeJS it would be good to provide similar context. We have legacy functions that rely on some of the data (such as awsRequestId etc).
I found some of the variables are injected into headers (such as `x-amzn-requestid`).

### What is the feature you are proposing to solve the problem?

Making it more compatible with AWS NodeJS implementation.

**Comment by david-wb** [comment]
Yes, the bun lambda runtime must forward the context object to the handler. Currently it only transforms the event into a bun `Request`, but the context needs to be passed as well. The context object may contain critical information such as results from a custom API Gateway Lambda Authorizer. See [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html#context-variable-reference).

**Comment by dreamorosi** [comment]
Hi @Jarred-Sumner, have taken a look at this and I think it'd be a good idea to pass the context object to the customer code. 

Looking at the code, I think I have a good idea on how to do this, and the changes would be limited only to the `packages/bun-lambda/runtime.ts` file, specifically [here](https://github.com/oven-sh/bun/blob/main/packages/bun-lambda/runtime.ts#L476-L478).

To provide backward compatibility with existing code running on Lambda that relies on the context object, the best way imo would be to attach a `context` object to the `Request` object, much like you're already doing with the original event under the undocumented `request.aws` property.

```diff
// @ts-ignore: Attach the original event to the Request
request.aws = event;
+  // @ts-ignore: Attach context to the Request
+  request.context = {
+    callbackWaitsForEmptyEventLoop: true,
+    functionVersion: process.env.AWS_LAMBDA_FUNCTION_VERSION,
+    functionName: process.env.AWS_LAMBDA_FUNCTION_NAME,
+    memoryLimitInMB: process.env.AWS_LAMBDA_FUNCTION_MEMORY_SIZE,
+    logGroupName: process.env.AWS_LAMBDA_LOG_GROUP_NAME,
+    logStreamName: process.env.AWS_LAMBDA_LOG_STREAM_NAME,
+    invokedFunctionArn: functionArn ,
+    awsRequestId: requestId,
+    getRemainingTimeInMillis: () => deadlineMs as number - Date.now(),
+  };
```

Then customers can access it in their handler like this:

```ts
import type { Context } from "aws-lambda";

export default {
  async handler(request: Request & { context: Context }): Promise<Response> {
    // @ts-ignore
    logger.info("Event", { event: request.aws });

    logger.info("Time remaining", {
      timeRemaining: request.context.getRemainingTimeInMillis(),
    });

    return new Response("Hello from Lambda!", {
      status: 200,
      headers: {
        "Content-Type": "text/plain",
      },
    });
  },
};
```

Ideally I think the best option, since the `aws` key seems to be undocumented, would be to namespace the whole thing and instead change the `formatRequest()` method like this:

```diff
-   // @ts-ignore: Attach the original event to the Request
-   request.aws = event;
+  // @ts-ignore Attach the aws object to the Request
+  request.aws = { event };
+  // @ts-ignore: Attach context to the Request
+  request.aws.context = {
+    callbackWaitsForEmptyEventLoop: true,
+    functionVersion: process.env.AWS_LAMBDA_FUNCTION_VERSION,
+    functionName: process.env.AWS_LAMBDA_FUNCTION_NAME,
+    memoryLimitInMB: process.env.AWS_LAMBDA_FUNCTION_MEMORY_SIZE,
+    logGroupName: process.env.AWS_LAMBDA_LOG_GROUP_NAME,
+    logStreamName: process.env.AWS_LAMBDA_LOG_STREAM_NAME,
+    invokedFunctionArn: functionArn ,
+    awsRequestId: requestId,
+    getRemainingTimeInMillis: () => deadlineMs as number - Date.now(),
+  };
```

But I don't know the project's backwards compatibility policy on this adapter, so the decision it's up to you really.

Either way I think it'd be great to have this exposed to customers as it would allow them to not have to implement the boilerplate code of having to fetch the values from env or maintain their own fork of the `bun-lambda` package just to run the runtime with `event` & `context`.

This is also not the only issue about this, #6378 would also be resolved with this change.

If you agree with one of the proposals, I'd be happy to send a PR. I have already this working on a local fork.
